packages/flutter_tools/lib/src/compile.dart

```diff
@@ -650,7 +646,7 @@ class DefaultResidentCompiler implements ResidentCompiler {
       sdkRoot,
       '--incremental',
       '--target=$targetModel',
-      '-Ddart.developer.causal_async_stacks=$causalAsyncStacks',
+      '-Ddart.developer.causal_async_stacks=${buildMode == BuildMode.debug}',
```

Comment:
This leads me to notice that we need to unify the flags used here with the flags used in `fuchsia_kernel_compiler.dart` either by keeping them in sync manually or by eliminating `fuchsia_kernel_compiler.dart` by merging its functionality into this file.

Reply:
It looks like the kernel_compiler and frontend_server support a slightly different set of flags and have slightly different output behavior. Is there any reason we can't standardize on the frontend_server instead?

Reply:
Ahh right. Functionality of the Fuchsia kernel compiler would need to be integrated into the frontend_server, so there's probably not much we can do right now other than manually try to keep the flags in-sync.